modular_skyrat/modules/liquids/code/liquid_systems/liquid_effect.dm

```diff
@@ -503,6 +513,32 @@
 	RegisterSignal(my_turf, COMSIG_ATOM_ENTERED, .proc/movable_entered)
 	RegisterSignal(my_turf, COMSIG_TURF_MOB_FALL, .proc/mob_fall)
 
+/**
+ * Handles COMSIG_PARENT_EXAMINE for the turf.
+ *
+ * Adds reagent info to examine text.
+ * Arguments:
+ * * source - the turf we're peekin at
+ * * examiner - the user
+ * * examine_text - the examine list
+ *  */
+/obj/effect/abstract/liquid_turf/proc/examine_turf(turf/source, mob/examiner, list/examine_text)
+	SIGNAL_HANDLER
+
+	// This should always have reagents if this effect object exists, but as a sanity check...
+	if(!length(reagent_list))
+		return
+
+	if(examiner.can_see_reagents()) //Show each individual reagent
+		examine_text += "<hr>"
+		examine_text += "\The [source] [liquid_state_messages["[liquid_state]"]]:"
+		for(var/datum/reagent/current_reagent as anything in reagent_list)
+			var/volume = reagent_list[current_reagent]
+			examine_text += "[round(volume, 0.01)] units of [initial(current_reagent.name)]"
```

Comment:
Did you test this with multiple reagents? It might look odd with no linebreak whatsoever, nor commas.
I'd probably suggest doing something like:
```
The iron floor is covered by a puddle of various reagents. (non-scanner version)

The iron floor is covered by a puddle of 105.3u of Water. (When you have the ability to see reagents, and there's only one)

The iron floor is covered by a puddle of the following reagents:
 - 105.3u of Water
 - 420.69u of Blood
(Version of reagent-scanning mode where you get details on each reagent)
```

Reply:
It currently uses the exact same format as looking at a reagent container item, ie basically the last one you listed for the scanner version without the actual bullets because the game doesn't do that for anything reagent related. Regarding line breaks, examine text is done as a list where each item is a line break, it's not one giant text string.